service/src/main/java/org/sunbird/service/systemsettings/SystemSettingsService.java

```diff
@@ -82,4 +89,44 @@ public <T> T getSystemSettingByFieldAndKey(
     }
     return null;
   }
+
+
+
+
+  public JsonNode getSystemSettingV2ByKey(String key, RequestContext context) {
+    logger.info("......................");
+    try {
+      DataCacheHandler.getConfigSettings().get(key);
+      String value = DataCacheHandler.getConfigSettings().get(key);
+      SystemSetting setting = new SystemSetting();
+      JsonNode responseJson = objectMapper.createObjectNode();
+      if (value != null) {
+//      setting = new SystemSetting(key, key, value);
+        ((ObjectNode) responseJson).put(JsonKey.ID, key);
+        ((ObjectNode) responseJson).put(JsonKey.FIELD, key);
+        ((ObjectNode) responseJson).put(JsonKey.VALUE, objectMapper.readTree(value));
+      } else {
+        setting = systemSettingDaoImpl.readByField(key, context);
+        ((ObjectNode) responseJson).put(JsonKey.ID, key);
+        ((ObjectNode) responseJson).put(JsonKey.FIELD, key);
+        ((ObjectNode) responseJson).put(JsonKey.VALUE, objectMapper.readTree(setting.getValue()));
+        if (null == setting) {
```

Comment:
this suppose to be 1st line inside else case

Reply:
done